### ISSUE TITLE
dyld: Library not loaded:@rpath/MapboxMobileEvents.framework/MapboxMobileEvents

### ISSUE THREAD

**Issue by lcf001** [mapbox/mapbox-gl-native-ios#131]
### Steps to reproduce
I compiled the SDK according to the instructions on the INSTALL.md and wrote a demo. When I run the demo I get the following error:
 dyld: Library not loaded:@rpath/MapboxMobileEvents.framework/MapboxMobileEvents
So how to add this library if I compile the SDK myself.

### Configuration

**Mapbox SDK versions:** 
**iOS/macOS versions:** 
**Device/simulator models:**
**Xcode version:**

**Comment by julianrex** [comment]
@lcf001 can you please double check that the `MapboxMobileEvents` framework is embedded in your application? Check the "Frameworks, Libraries and Embedded Content" section of your application target.

It should look something like this:

<img width="706" alt="Screen Shot 2020-01-06 at 3 30 13 PM" src="https://user-images.githubusercontent.com/3765757/71846904-a2f02c80-3099-11ea-8c1f-4b9546c8667f.png">

**Comment by lcf001** [comment]
I can't find  the MapboxMobileEvents framework。In which folder is it when I compile the sdk?

**Comment by lcf001** [comment]
> @lcf001 can you please double check that the `MapboxMobileEvents` framework is embedded in your application? Check the "Frameworks, Libraries and Embedded Content" section of your application target.
> 
> It should look something like this:
> 
> <img alt="Screen Shot 2020-01-06 at 3 30 13 PM" width="706" src="https://user-images.githubusercontent.com/3765757/71846904-a2f02c80-3099-11ea-8c1f-4b9546c8667f.png">

I can't find the MapboxMobileEvents framework。In which folder is it when I compile the sdk?

**Comment by lcf001** [comment]
@julianrex I can't find the MapboxMobileEvents framework。In which folder is it when I compile the sdk?

**Comment by julianrex** [comment]
`MapboxMobileEvents.framework` was recently added as a dependency. Please see the releases notes: https://github.com/mapbox/mapbox-gl-native-ios/releases/tag/ios-v5.6.0

**Comment by lcf001** [comment]
@julianrex Do I download mapbox-ios-sdk-5.6.0-dynamic-with-events.zip separately? but it is too slow. Can I remove this dependency？

**Comment by julianrex** [comment]
@lcf001 how are you integrating the Maps SDK? Manually, through Cocoapods or Carthage? 

If manually, you should download `mapbox-ios-sdk-5.6.0-dynamic-with-events.zip` in preference over any other. The `MapboxMobileEvents.framework` is required, i.e. you cannot remove it.

For reference, this is not new code - it's been extracted from the Maps SDK to address some issues and to remove duplicated code.

I'm going to close this issue as answered.

**Comment by falnatsheh** [comment]
I had the same issue when using Carthage, I believe the integration instructions need to be updated to include:

> Then add the framework’s path to the Input Files for this script phase:
```
$(SRCROOT)/Carthage/Build/iOS/Mapbox.framework
$(SRCROOT)/Carthage/Build/iOS/MapboxMobileEvents.framework   ◀️ 
```

**Comment by 1ec5** [comment]
Thank you, we’ll update [this page](https://www.mapbox.com/install/ios/carthage-build-setup/) shortly with the change you’ve suggested.

**Comment by jordanderson** [comment]
I think [this page](https://www.mapbox.com/install/ios/download/) also needs to be updated to include `MapboxMobileEvents.framework`.

**Comment by jsgv** [comment]
I am installing with Cocoapods, but still receiving this error. Docs do not specify any special instructions regarding the framework using the Cocoapods method.

**Comment by anfriis** [comment]
> I am installing with Cocoapods, but still receiving this error. Docs do not specify any special instructions regarding the framework using the Cocoapods method.
> 
> **Edit:**
> `use_frameworks!` needs to be set in Podfile in order to work.

This is an unfortunate change IMO @julianrex.
The `use_frameworks!` flag will cause all pods in a project to be linked as dynamic frameworks which tend to make your app launch slower and increase in bundle size compared to using static libraries ([link 1](https://medium.com/@acecilia/static-vs-dynamic-frameworks-in-swift-an-in-depth-analysis-ff61a77eec65)) ([link 2](https://blog.automatic.com/how-we-cut-our-ios-apps-launch-time-in-half-with-this-one-cool-trick-7aca2011e2ea)). Apple also suggests going away from dynamic frameworks. 
Furthermore it can cause build errors in apps that has other Pods that cannot be linked as a  dynamic frameworks e.g. [react-native-firebase](https://github.com/invertase/react-native-firebase/issues/3024#issuecomment-577365446), which also causes my app to throws errors when building.

**Comment by julianrex** [comment]
@anfriis thanks for the comment. Building from source and/or providing a static framework is an important use-case and is being considered (see also https://github.com/mapbox/mapbox-gl-native-ios/issues/149).

**Comment by systemlevel** [comment]
Running into the same wall as @anfriis has called out here.  This is unfortunate.  

We are stuck without a static framework being provided for similar reasons @anfriis has pointed out.  Is there a reason why we are requiring the use of use_frameworks! when Apple is discouraging dynamic frameworks moving forward?  

https://github.com/react-native-mapbox-gl/maps/issues/635

**Comment by like-milk** [comment]
> @lcf001 can you please double check that the `MapboxMobileEvents` framework is embedded in your application? Check the "Frameworks, Libraries and Embedded Content" section of your application target.
> 
> It should look something like this:
> 
> <img alt="Screen Shot 2020-01-06 at 3 30 13 PM" width="706" src="https://user-images.githubusercontent.com/3765757/71846904-a2f02c80-3099-11ea-8c1f-4b9546c8667f.png">

It worked! Thank you very much!

**Comment by swapnizzle** [comment]
I encountered this issue when trying to build to my actual iOS device (but it worked fine in the simulator). I spent at least 10 hours trying to fix this bug. Tried everything in all the forums, nothing worked. But this is what finally worked for me!

- Upgrading from iOS 13.3.1 to iOS 13.4.1

I couldn't have imagined this, but its true that <b> iOS 13.3.1 breaks the third parties signature unless you have a paid developer account!</b>
